### PR TITLE
Retiring as PythonPackage maintainer

### PR DESCRIPTION
**Pull request by adamjstewart**
I have served as the maintainer for every single Python library in Spack for over 3 years now (#21125). Unfortunately, it's time to step down.

### Why now?

* Too much work: the number of Python packages in Spack has nearly doubled in the last 3 years, from 1.3k to 2.5k, and is too much for a single (sane) person to maintain
* Not enough time: I would like to spend more time on my postdoc and [my other baby](https://github.com/microsoft/torchgeo)

### Can we still ask you to review PRs or solve issues?

Yes! Please ping me for the following:

1. Any machine learning package
2. Any geospatial package
3. Any package on which I am explicitly listed as a maintainer
4. Any time there is a question/confusion regarding the proper way to package any Python library

Please refrain from pinging me for the following:

5. Any simple Python package update or new package that does not fall under the previous categories

### So should we just merge everything blindly?

I would prefer that someone at least _attempts_ to review Python PRs before merging. 99% of Python packages carefully list their dependencies; all you have to do is make sure that list matches the package. https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html contains instructions for where to find these. I would be more than happy to teach others how to review Python PRs if anyone has any interest in this.

### Will you ever come back?

Maybe. @haampie and others are doing some great work on automating the 99% of packages that do not require manual attention to correct known issues or document non-Python dependencies. I have high hopes that a future in which most builtin Python/R/Perl packages will be removed from Spack entirely and built on-the-fly. If this happens, depending on where I'm at in my career, I may be willing to reprise my role. 

Let me know if you have any other questions!

### CODE REVIEW

**Comment by haampie** [comment]
:( sad to see this. Thanks a lot for all your work that went into Spack.

```
lib/spack/spack/build_systems/python.py:30: [F401] 'spack.directives.maintainers' imported but unused
```

somewhat ironic

**Comment by skosukhin** [comment]
I'm sorry, I can't approve this PR :(

**Comment by alalazo** [comment]
It seems even unit tests are assuming you are THE Python maintainer!  :slightly_smiling_face: Thank a lot for all the work, I'll fix the unit tests and merge the PR!

**Comment by tldahlgren** [comment]
Aw.  I've enjoyed "tag teaming" python PRs with you 😉 .  Your maintainer contributions will be missed.

**Review by alalazo — APPROVED** [review]
I'm tempted to say we can't merge this with failing coverage... :stuck_out_tongue_winking_eye:

**Comment by tldahlgren** [comment]
Merged?!  Nooooooooo.  😭

**Comment by tgamblin** [comment]
@adamjstewart lemme just join in and say thanks or making Spack what it is today. Here's the contributor graph of all the lines in Spack packages by author. Adam's been the top package contributor since 2017 or earlier. We wouldn't have X11, many of the ML packages, the Python ecosystem, or many more parts of Spack without him.

Also, he contributed a lot of these things at times when the package ecosystem was much smaller than it is today. These days, you can expect that many of your dependencies are already in Spack. When @adamjstewart started, you couldn't, and it was people like him who lowered the barrier for everyone else by adding all the fundamentals.

I am not sure how Adam got a Ph.D. while also doing all of that but somehow he's managed, and now he's got other things to worry about. Thanks everyone for wishing him well!

So thanks @adamjstewart -- hope to see you around on the ML packages because even that is still a monumental task. And we'll double down on automating and simplifying Python (and other ecosystem) management in the meantime.

loc-in-packages-by-author

**Pull request merged**
